packages/salesforcedx-vscode-core/src/commands/baseDeployRetrieve.ts

```diff
@@ -63,7 +63,7 @@ export abstract class DeployRetrieveExecutor<
 
     try {
       const components = await this.getComponents(response);
-      this.setApiVersionOn(components);
+      void this.setApiVersionOn(components);
```

Comment:
just curious what lint error are you working around here?

Reply:
also this void hurts my brain....what is it doing?

Reply:
Ok so I pulled it local b/c short on time, but it looks like setApiVersionOn is async.  Are we purposely not waiting on the promise to finish before continuing on? Unless we're purposefully not waiting on the async method to finish before moving on (and if we are we should add a code comment about why) We should just use `await` here vs `void`

Reply:
When I don't use void here, I get this linter error: "Promises must be handled appropriately"

When I use await here, all the tests in the file fail with "Failure in test '[object Object]': AuthInfoCreationError: Must pass a username and/or OAuth options when creating an AuthInfo instance."

I'll switch to await and see if I can figure it why it makes the other tests fail.